src/vm/eventpipebuffermanager.cpp

```diff
@@ -32,6 +32,23 @@ EventPipeBufferManager::EventPipeBufferManager()
 #endif // _DEBUG
 }
 
+EventPipeBufferManager::~EventPipeBufferManager()
+{
+    CONTRACTL
+    {
+        THROWS;
+        GC_TRIGGERS;
+        MODE_ANY;
+    }
+    CONTRACTL_END;
+
+    if(m_pPerThreadBufferList != NULL)
```

Comment:
If you're going to delete the list, you should also delete the list items (though not the buffers themselves - these could possibly be in-use).

Reply:
I've added deletion of the list items too.